### ISSUE TITLE
Add ToFormData class

### ISSUE THREAD

**Issue by fizruk** [GetShopTV/swagger2#30]
Form data is halfway between `ParamSchema` and `Schema` with an extra property that it can handle `SwaggerFile`. I think it makes sense to introduce a class `ToFormData` for that. Instances of this class can be derived generically for records where each field has a `ParamSchema` instance.

The correspondence of type-level and value-level classes is this:
- `ToParamSchema` — `ToHttpApiData`;
- `ToFormData` — `ToFormUrlEncoded`;
- `ToSchema` — `ToJSON`.

Having `ToFormData` class with `Generic`-based deriving should reduce a significant amount of boilerplate.

The problem I currently have with `ToFormData` class is that `swagger2` model does not allow for this class to be pretty:

```haskell
type FormDataParam =
  ( ParamSchema ParamOtherSchema
  , Maybe Bool -- required?
  , Maybe Bool -- allow empty values?
  )

class ToFormData a where
  toFormData :: proxy a -> HashMap ParamName FormDataParam
```

Basically, `FormDataParam` should be just `Param` with the restriction that `ParamLocation` is `ParamFormData`. But we cannot specify that constraint in current model.
Actually I don't know if we can specify it at all (without resorting to existentials like `SomeParam`).

For now we can define a data type for form data parameters together with conversion function:

```haskell
data FormDataParam = FormDataParam
  { formDataParamDescription     :: Maybe Text
  , formDataParamRequired        :: Maybe Bool
  , formDataParamAllowEmptyValue :: Maybe Bool
  , formDataParamParamSchema     :: ParamSchema ParamOtherSchema
  }

formDataParamToParam :: FormDataParam -> Param
formDataParamToParam = ...
```

@dmjio what do you think about this hacky solution?

**Comment by fizruk** [comment]
Forms are now in `http-api-data-0.3`!
